hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeAttributeProvider.java

```diff
@@ -17,19 +17,227 @@
+        this.path = val;
+        return this;
+      }
+
+      public Builder ancestorIndex(int val) {
+        this.ancestorIndex = val;
+        return this;
+      }
+
+      public Builder doCheckOwner(boolean val) {
+        this.doCheckOwner = val;
+        return this;
+      }
+
+      public Builder ancestorAccess(FsAction val) {
+        this.ancestorAccess = val;
+        return this;
+      }
+
+      public Builder parentAccess(FsAction val) {
+        this.parentAccess = val;
+        return this;
+      }
+
+      public Builder access(FsAction val) {
+        this.access = val;
+        return this;
+      }
+
+      public Builder subAccess(FsAction val) {
+        this.subAccess = val;
+        return this;
+      }
+
+      public Builder ignoreEmptyDir(boolean val) {
+        this.ignoreEmptyDir = val;
+        return this;
+      }
+
+      public Builder operationName(String val) {
+        this.operationName = val;
+        return this;
+      }
+
+      public Builder callerContext(CallerContext val) {
+        this.callerContext = val;
+        return this;
+      }
+    }
+
+    public AuthorizationContext(
+        String fsOwner,
+        String supergroup,
+        UserGroupInformation callerUgi,
+        INodeAttributes[] inodeAttrs,
+        INode[] inodes,
+        byte[][] pathByNameArr,
+        int snapshotId,
+        String path,
+        int ancestorIndex,
+        boolean doCheckOwner,
+        FsAction ancestorAccess,
+        FsAction parentAccess,
+        FsAction access,
+        FsAction subAccess,
+        boolean ignoreEmptyDir) {
+      this.fsOwner = fsOwner;
+      this.supergroup = supergroup;
+      this.callerUgi = callerUgi;
+      this.inodeAttrs = inodeAttrs;
+      this.inodes = inodes;
+      this.pathByNameArr = pathByNameArr;
+      this.snapshotId = snapshotId;
+      this.path = path;
+      this.ancestorIndex = ancestorIndex;
+      this.doCheckOwner = doCheckOwner;
+      this.ancestorAccess = ancestorAccess;
+      this.parentAccess = parentAccess;
+      this.access = access;
+      this.subAccess = subAccess;
+      this.ignoreEmptyDir = ignoreEmptyDir;
+    }
+
+    public AuthorizationContext(
+        String fsOwner,
+        String supergroup,
+        UserGroupInformation callerUgi,
+        INodeAttributes[] inodeAttrs,
+        INode[] inodes,
+        byte[][] pathByNameArr,
+        int snapshotId,
+        String path,
+        int ancestorIndex,
+        boolean doCheckOwner,
+        FsAction ancestorAccess,
+        FsAction parentAccess,
+        FsAction access,
+        FsAction subAccess,
+        boolean ignoreEmptyDir,
+        String operationName,
+        CallerContext callerContext) {
+      this(fsOwner, supergroup, callerUgi, inodeAttrs, inodes,
+          pathByNameArr, snapshotId, path, ancestorIndex, doCheckOwner,
+          ancestorAccess, parentAccess, access, subAccess, ignoreEmptyDir);
+      this.operationName = operationName;
```

Comment:
can we have only one constructor with the all the parameters to avoid the operationName/callerContext assign in multiple places?

Reply:
removed this constructor. It is not used.